flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutorGateway.java

```diff
@@ -216,9 +227,18 @@
 	 */
 	CompletableFuture<Boolean> canBeReleased();
 
+
+	/**
+	 * Requests for the historical log file names on the TaskManager.
+	 *
+	 * @return A Tuple2 Array with all log file names with its length.
+	 */
+	CompletableFuture<Collection<Tuple2<String, Long>>> requestLogList(@RpcTimeout Time timeout);
```

Comment:
I would prefer not to use `Tuple2<String, Long>`. IMO tuple types should be used sparingly because its semantics are often not clear. I would prefer to introduce a new type or just use `org.apache.flink.runtime.rest.messages.taskmanager.LogInfo`. There are some downsides with reusing types but it has been done like that before (for example, see `ResourceManager#requestTaskManagerInfo()`)

Reply:
If the reuse type is ok, I'll use it.